AK/FloatingPoint.h

```diff
@@ -13,96 +13,75 @@
 namespace AK {
 
 template<typename T>
-struct FloatExtractor;
+struct FloatExtractorTraits;
 
 #ifdef AK_HAS_FLOAT_128
 template<>
-struct FloatExtractor<f128> {
-    static constexpr FloatExtractor<f128> from_float(f128 f) { return bit_cast<FloatExtractor<f128>>(f); }
-    constexpr f128 to_float() const { return bit_cast<f128>(*this); }
-
+struct FloatExtractorTraits<f128> {
     using ComponentType = unsigned __int128;
     static constexpr int mantissa_bits = 112;
-    static constexpr ComponentType mantissa_max = (((ComponentType)1) << 112) - 1;
-    static constexpr int exponent_bias = 16383;
     static constexpr int exponent_bits = 15;
-    static constexpr unsigned exponent_max = 32767;
-
-    ComponentType mantissa : 112;
-    ComponentType exponent : 15;
-    ComponentType sign : 1;
 };
-// Validate that f128 and the FloatExtractor struct are 128 bits.
-static_assert(AssertSize<f128, 16>());
-static_assert(AssertSize<FloatExtractor<f128>, sizeof(f128)>());
 #endif
 
 #ifdef AK_HAS_FLOAT_80
 template<>
-struct FloatExtractor<f80> {
-    static constexpr FloatExtractor<f80> from_float(f80 f) { return bit_cast<FloatExtractor<f80>>(f); }
-    constexpr f80 to_float() const { return bit_cast<f80>(*this); }
-
-    using ComponentType = unsigned long long;
-    static constexpr int mantissa_bits = 64;
-    static constexpr ComponentType mantissa_max = ~0ull;
-    static constexpr int exponent_bias = 16383;
-    static constexpr int exponent_bits = 15;
-    static constexpr unsigned exponent_max = 32767;
+struct FloatExtractorTraits<f80> {
+    using ComponentType = u64;
```

Comment:
This should likely be promoted to `unsigned __int128`, that would fix the compilation error